backend/pkg/kafka/deserializer.go

```diff
@@ -136,7 +136,25 @@ func (d *deserializer) deserializePayload(payload []byte, topicName string, reco
 		}
 	}
 
-	// 2. Test for valid XML
+	// 2. Test for json schema
+	if payload[0] == byte(0) {
+		schemaID := binary.BigEndian.Uint32(payload[1:5])
```

Comment:
I think this can end up in panics because a value might be a single zero byte. Line 141 would try to read byte 1-5 which would be out of bounds. For the avro check we use something like that before checking the magic byte. 

```go
if d.SchemaService != nil && len(payload) > 5
```

Reply:
done